### ISSUE TITLE
Documentation does not mention that directory traversal omits directories beginning with '.' 

### ISSUE THREAD

**Issue by jamesquilty** [markdownlint/markdownlint#342]
I've just discovered that `mdl <directory>` and `mdl --git-recurse <directory>` produce different output. While the documentation says "when mdl is given a directory name on the command line, it will recurse into that directory looking for markdown files to process," it appears that directories beginning with a `.` (dot) are omitted from the `mdl <directory>` traversal.

At the root level of my repository, `mdl .` does not traverse my `.gitlab/issue_templates/` directory and won't report formatting style violations on the but `mdl --git-recurse .` _will_ traverse `.gitlab/issue_templates/` directory and reports violations.

```shell
$ mdl .
# No output, all Markdown files are OK
$ mdl --git-recurse .
.gitlab/issue_templates/<template>.md:1: MD002 First header should be a top level header
.gitlab/issue_templates/<template>.md:1: MD041 First line in file should be a top level header

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

This behaviour is actually what I want, because templates should not be checked for formatting violations. The only confusing thing was that the documentation on the main page and <https://github.com/markdownlint/markdownlint/blob/master/docs/configuration.md> doesn't mention that `mdl <directory>` _won't_ descend into directories with names beginning with a `.` (dot). I think that this is a useful feature which should be documented 😄

**Comment by oliviertassinari** [comment]
Same here, I noticed this for ` .github/PULL_REQUEST_TEMPLATE.md`. How can we get these files linted?
